src/miniscript/context.rs

```diff
@@ -276,6 +276,8 @@ pub trait ScriptContext:
     /// 34/66 for Bare/Legacy based on key compressedness
     /// 34 for Segwitv0, 33 for Tap
     fn pk_len<Pk: MiniscriptKey>(pk: &Pk) -> usize;
+    /// Local helper function to display error messages with context
+    fn to_string() -> String;
```

Comment:
I think this could return a `&'static str` (and similarly the `String`s in all the error variants could be `&'static str`s). The function should probably be called `to_str` then.

Reply:
It is also not clear why we can't just require `ScriptContext: Display`, implement `Display` for specific context types and get rid from this method here